pkg/scheduler/framework/runtime/framework_test.go

```diff
@@ -1144,6 +1161,145 @@ func TestPostFilterPlugins(t *testing.T) {
 	}
 }
 
+func TestFilterPluginsWithNominatedPods(t *testing.T) {
+	podNominator := internalqueue.NewPodNominator()
+	tests := []struct {
+		name             string
+		preFilterPlugins []*TestPlugin
+		filterPlugins    []*TestPlugin
+		pod              *v1.Pod
+		nominatedPod     *v1.Pod
+		node             *v1.Node
+		nodeInfo         *framework.NodeInfo
+		wantStatus       *framework.Status
+	}{
+		{
+			name:             "node has no nominated pod",
+			preFilterPlugins: []*TestPlugin{},
+			filterPlugins:    []*TestPlugin{},
+			pod:              lowPriorityPod,
+			nominatedPod:     nil,
+			node:             node,
+			nodeInfo:         framework.NewNodeInfo(pod),
+			wantStatus:       nil,
+		},
+		{
+			name: "node has greater priority nominated pod and all filters succeed",
+			preFilterPlugins: []*TestPlugin{
+				{
+					name: "TestPlugin1",
+					inj: injectedResult{
+						PreFilterAddPodStatus: int(framework.Success),
+					},
+				},
+			},
+			filterPlugins: []*TestPlugin{
+				{
+					name: "TestPlugin2",
+					inj: injectedResult{
+						FilterStatus: int(framework.Success),
+					},
+				},
+			},
+			pod:          lowPriorityPod,
+			nominatedPod: highPriorityPod,
+			node:         node,
+			nodeInfo:     framework.NewNodeInfo(pod),
+			wantStatus:   nil,
+		},
+		{
+			name: "node has greater priority nominated pod and pre filters fail",
+			preFilterPlugins: []*TestPlugin{
+				{
+					name: "TestPlugin1",
+					inj: injectedResult{
+						PreFilterAddPodStatus: int(framework.Error),
+					},
+				},
+			},
+			filterPlugins: []*TestPlugin{},
+			pod:           lowPriorityPod,
+			nominatedPod:  highPriorityPod,
+			node:          node,
+			nodeInfo:      framework.NewNodeInfo(pod),
+			wantStatus:    framework.NewStatus(framework.Error, `running AddPod on PreFilter plugin "TestPlugin1": injected status`),
+		},
+		{
+			name: "node has greater priority nominated pod and filters fail",
+			preFilterPlugins: []*TestPlugin{
+				{
+					name: "TestPlugin1",
+					inj: injectedResult{
+						PreFilterAddPodStatus: int(framework.Success),
+					},
+				},
+			},
+			filterPlugins: []*TestPlugin{
+				{
+					name: "TestPlugin2",
+					inj: injectedResult{
+						FilterStatus: int(framework.Error),
+					},
+				},
+			},
+			pod:          lowPriorityPod,
+			nominatedPod: highPriorityPod,
+			node:         node,
+			nodeInfo:     framework.NewNodeInfo(pod),
+			wantStatus:   framework.NewStatus(framework.Error, `running "TestPlugin2" filter plugin for pod "": injected filter status`),
+		},
+	}
```

Comment:
Can you add a test with the pod as high-priority, and nomiantedPod as low-priority. And PreFilter returns `Unschedulable` status, the expected result should be `Success` as PreFilter won't be called.

Reply:
where does PreFilter get invoked? I don't see a call to RunPreFilterPlugins in this code path, what am I missing?

Reply:
> where does PreFilter get invoked?

I should say RunPreFilterExtensions, it's in the first attempt (i=0), it tries to incorporate low-priority nominated pod:

https://github.com/kubernetes/kubernetes/blob/dc39ab2417bfddcec37be4011131c59921fdbe98/pkg/scheduler/core/generic_scheduler.go#L390-L393